pull/Java8TASK/files/ExtraTrial.java

```diff
@@ -0,0 +1,37 @@
+public class ExtraTrial extends Trial implements Cloneable{
+    public int mark3;
+    public static final int PASS_VALUE_EXTRA = 85;
+
+    public ExtraTrial(String account, int mark1, int mark2, int mark3) {
+        super(account, mark1, mark2);
+        if (mark3 < 0 || mark3 > 100) {
+            throw new IllegalArgumentException("Mark 3 is not valid. Mark values must be between 0 and 100 inclusive");
+        } else {
+            this.mark3 = mark3;
+        }
+    }
+
+    public ExtraTrial(ExtraTrial obj) {
+        super(obj.account, obj.mark1, obj.mark2);
+        mark3 = obj.mark3;
+    }
+    @Override
+    public ExtraTrial clone() {
+        return new ExtraTrial(this);
+    }
+
+    @Override
+    public boolean isPassed() {
+        return super.isPassed() && mark3 > PASS_VALUE_EXTRA;
+    }
+
+    @Override
+    public void clearAllMarks() {
+        super.clearAllMarks();
+        mark3 = 0;
+    }
+    @Override
+    public String toString() {
+        return String.format("%s;%d;%d;%d;%d;%b", account, mark1, mark2, mark3, PASS_VALUE_EXTRA, isPassed());

```

Comment:
```suggestion
        return String.format("%s;%d;%d;%d;%d;%b", account, mark1, mark2, mark3, isPassed());
```
имеет смысл через `fieldsToString`.